crates/solidity/src/project/contract/mod.rs

```diff
@@ -154,6 +167,19 @@ impl Contract {
             )
         })?;
 
+        if let Some(dinfo) = context.debug_info() {
+            dinfo.finalize_module()
+        }
+
+        #[cfg(debug_assertions)]
+        if let Err(err) = context.verify() {
+            eprintln!("Invalid LLVM module generated:");
```

Comment:
FIY error context should be added using the anyhow context. In stderr it might also be swallowed in recursive calls.

Reply:
Fixed.

Reply:
Fixed